Add scripted build and run for different miniqmc binaries

jplehr opened the pull request:
The script is pretty minimal in terms of configurability and error checking, but should be a reasonable starting point.

jplehr committed:
[MiniQMC] Adds run script for MiniQMC

Initial version of a run script for the miniqmc proxy.

The script runs different executables produced from the build
1 check_spo_batched_reduction
2 check_spo
3 miniqmc

jplehr committed:
More flexibility in the script

jplehr committed:
Adds documentation and git clone step

jplehr committed:
Updates to run_miniQMCPack.sh

Adds -fopenmp-assume-no-nested-parallelism switch to work around current issue
in AOMP when -fdisable-host-devmem is given.

jplehr committed:
Fixes miniqmc git SHA by default

dpalermo commented on bin/run_miniQMCPack.sh:
```diff
@@ -0,0 +1,72 @@
+export rocsolver_DIR=${ROCM_INSTALL_PATH}/lib/cmake/rocsolver/
+
+# Set the default build prefix, i.e., build-top-level
+: ${MQMCPACK_BUILD_PREFIX:=$HOME/miniqmc_build}
```
Might want to default to $AOMP_REPOS_TEST/miniqmc_build rather than the users home directory.  Using AOMP_REPOS_TEST will match other run scripts.

dpalermo commented on bin/run_miniQMCPack.sh:
```diff
@@ -0,0 +1,72 @@
+# --- end standard header ----
+
+# Default ROCm installation
+: ${ROCM_INSTALL_PATH:=/opt/rocm-5.3.0}
```
Rename to just ROCM (since we aren't installing ROCm in this script)...and default to /opt/rocm (no version number...which is a symlink to the installed version)

dpalermo commented on bin/run_miniQMCPack.sh:
```diff
@@ -0,0 +1,72 @@
+# Set the default build directory name
+: ${MQMCPACK_BUILD_DIR:=${MQMCPACK_BUILD_PREFIX}/build_aomp_clang}
+# Path to the miniqmc source directory
+: ${MQMC_SOURCE_DIR:=$HOME/miniqmc_src}
```
Might want to default to $AOMP_REPOS_TEST/miniqmc_src rather than the users home directory. Using AOMP_REPOS_TEST will match other run scripts.

dpalermo commented on bin/run_miniQMCPack.sh:
```diff
@@ -0,0 +1,72 @@
+
+# run_miniQMCPack.sh - runs several MiniQMC binaries
+# The user should set at least these two variables:
+# MQMC_BUILD_PREFIX       miniqmc build prefix (default: ~/miniqmc_build)
```
Move these env vars to the "user can set" section.  No env vars need to be set to run this script.

ye-luo commented on bin/run_miniQMCPack.sh:
```diff
@@ -0,0 +1,72 @@
+#! /usr/bin/env bash
+
+# run_miniQMCPack.sh - runs several MiniQMC binaries
```
It is simply `miniQMC` not `miniQMCPack`;)

jplehr committed:
Address reviewer comments

- Rename things to remove "pack" from all the variables
- Moves things to AOMP_REPOS_TEST instead of HOME
- Rename to ROCM instead of ROCM_INSTALL_DIR
